.gitlab-ci.yml

```
@@ -108,16 +122,53 @@ build:
   script:
     - cargo build --no-default-features --target wasm32-unknown-unknown --verbose
 
+
+#### stage:                        tag
+# this stage will only create a tag in the repo, not release
+
+tag-job:
+  stage: tag
+  <<:                              *kubernetes-env
+  <<:                              *vault-secrets
+  <<:                              *publish-refs
+  script:
+    - export CURRENT_TAG=$(git describe --tags --abbrev=0)
+    - export PKG_VER=v$(cat Cargo.toml | grep -A 5 package] | grep version | cut -d '=' -f 2 | tr -d '"' | tr -d " ")
+    - echo $CURRENT_TAG $PKG_VER
+    - git config user.name "${GITHUB_USER}"
+    - git config user.email "devops-team@parity.io"
+    - git config remote.origin.url "https://${GITHUB_TOKEN}@github.com/paritytech/${CI_PROJECT_NAME}.git"
+    - git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+    - if [ $CURRENT_TAG == $PKG_VER  ];
+        then
+          echo "Tag is the same. Nothing to do";
+          export TAG=old;
+        else
+          echo "Tag has changed";
+          git tag -a $PKG_VER -m "new tag";
+          git tag --list;
+          git push origin --tags;
+          export TAG=new;
+        fi
+    - echo "TAG=$TAG" > tag.env;
+  artifacts:
+    reports:
+      dotenv: tag.env
+
 #### stage:                       publish
 
 unleash-to-crates-io:
   stage:                           publish
-  <<:                              *docker-env
+  <<:                              *kubernetes-env
```

better keep Rust jobs to docker executors
```suggestion
  <<:                              *docker-env
```